File: 프로그래머스/0/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.js
function add(arr, num) {
    for(let i = 0; i < num*2; i++) {
        arr.push(num)
    }
}
function del(arr, num) {
    for(let i = 0; i< num; i++) {
        arr.pop()
    }
}
function solution(arr, flag) {
    var answer = [];
    
    for(let i = 0; i < arr.length; i++) {
        const num = arr[i];
        const isTure = flag[i];
        if(isTure) add(answer, num);
        else del(answer, num)
  
    }
    return answer;
}

File: 프로그래머스/0/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.py
def solution(arr, flag):
    result = []
    for i in range(len(flag)):
        if flag[i] == True:
            for _ in range(arr[i] * 2):
                result.append(arr[i])
        if flag[i] == False:
            for _ in range(arr[i]):
                result.pop()
    return result